libraries/Matter/src/Matter.cpp

```diff
@@ -0,0 +1,158 @@
+#include <sdkconfig.h>
+#ifdef CONFIG_ESP_MATTER_ENABLE_DATA_MODEL
+
+#include <Matter.h>
+#include <app/server/Server.h>
+#include "MatterEndPoint.h"
+
+using namespace esp_matter;
+using namespace esp_matter::attribute;
+using namespace esp_matter::endpoint;
+using namespace chip::app::Clusters;
+
+constexpr auto k_timeout_seconds = 300;
+
+static bool _matter_has_started = false;
+static node::config_t node_config;
+static node_t *deviceNode = NULL;
+
+typedef void *app_driver_handle_t;
+esp_err_t matter_light_attribute_update(
+  app_driver_handle_t driver_handle, uint16_t endpoint_id, uint32_t cluster_id, uint32_t attribute_id, esp_matter_attr_val_t *val
+);
+
+// This callback is called for every attribute update. The callback implementation shall
+// handle the desired attributes and return an appropriate error code. If the attribute
+// is not of your interest, please do not return an error code and strictly return ESP_OK.
+static esp_err_t app_attribute_update_cb(
+  attribute::callback_type_t type, uint16_t endpoint_id, uint32_t cluster_id,
+  uint32_t attribute_id, esp_matter_attr_val_t *val, void *priv_data)
+{
+  esp_err_t err = ESP_OK;
+  if (type == PRE_UPDATE) { /** Callback before updating the value in the database */

```

Comment:
Shouldn't this be a `switch` or `if/else if`?

Reply:
It could be a `switch{}`.
I have just added the other "cases" for future use.
I'll change it.

Reply:
Done.